src/Attributes/FilterDefaultField.php

```diff
@@ -1,12 +1,27 @@
 <?php
 
-declare(strict_types=1);
-
 namespace Drush\Attributes;
 
 use Attribute;
+use Consolidation\AnnotatedCommand\Parser\CommandInfo;
 
-#[Attribute(Attribute::TARGET_METHOD)]
-class FilterDefaultField extends \Consolidation\AnnotatedCommand\Attributes\FilterDefaultField
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_CLASS)]
+class FilterDefaultField
 {
+    const KEY = 'filter-default-field';
+
+    /**
+     * @param $field
+     *   A field name to filter on by default.
+     */
+    public function __construct(
+        public string $field
+    ) {
+    }
+
+    public static function handle(\ReflectionAttribute $attribute, CommandInfo $commandInfo)
+    {
+        $args = $attribute->getArguments();
+        $commandInfo->addAnnotation('filter-default-field', $args['field']);
```

Comment:
Here and elsewhere, `KEY` is defined in the class but not used in handle()

Reply:
Just pushed a fix with a couple tweaks. Depends on another PR in output-formatters https://github.com/consolidation/output-formatters/pull/113

Reply:
I have experimented with that approach earlier in this PR and can't decide whats best. Currently the validation is done right in execute() and isn't not DRY. See [ImageFlushCommand](https://github.com/drush-ops/drush/pull/6135/files#diff-3b7cd99de69343fefe9c81829f97ac88a82904e64b45cdda528de7974483518aR65-R101). Compare with https://github.com/drush-ops/drush/blob/f923a801dd50d4aa17fd2390790336bc962785a3/src/Listeners/ValidateModulesEnabledListener.php and its associated Attribute. IMO the current approach is less magical and I'm slightly valuing that these days.